tst/unit/test_tasklist.cpp

```diff
@@ -0,0 +1,50 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+// STL Includes
+#include <memory>
+
+// Third Party Includes
+#include <catch2/catch.hpp>
+
+// Internal Includes
+#include "basic_types.hpp"
+#include "tasks/task_list.hpp"
+
+using parthenon::TaskID;
+using parthenon::TaskList;
+using parthenon::TaskStatus;
+
+TEST_CASE("Task Object Lifecycle", "[TaskList][AddTask]") {
```

Comment:
So this test checks that tasks and objects within them are garbage collected appropriately. That's good. But it strikes me as an odd choice of a thing to test. I don't object, but can you explain your reasoning?

Reply:
Oh, the only reason I added this test was because of the original purpose of my changes before I realized it was going to be more work than I expected. I can delete it if it's not useful.

Reply:
Nah, leave it. It's better than nothing. And if this is something you're going to have to tackle eventually, might as well get the task in now.

Reply:
Oh, I'll probably stick a `std::shared_ptr` in the middle and call it good so I can still copy my move-only state...